### ISSUE TITLE
Python3 (again!)

### ISSUE THREAD

**Issue by stevemolloy** [Archman/beamline#8]
As I wrote for pyrpn, I was wondering if there was anything that prevented this being expanded to allow use with python3?
(As before, I'm more than willing to do the work myself if you don't see any major stumbling blocks.)

**Comment by zhangt58** [comment]
Compatibility could be resolved by using `__future__` module, but the GUI does depend on `wxPython`, which requires some non-trivial work to make it run under Python 3.

The most recent version of wxPython is `4.0.0b2` (beta version), which supports Python 3, but there are some issues need to be fixed. I was working to upgrade another project ([felapps](https://github.com/archman/felapps)) with this version of `wx` half years ago, there're lots of changes.

**Comment by smartsammler** [comment]
I created pull request #9 to have everything except the UI already transformed to Python 3 compatible code, as far as the tests and my daily usage showed.

**Comment by zhangt58** [comment]
>I created pull request #9 to have everything except the UI already transformed to Python 3 compatible code, as far as the tests and my daily usage showed.

Are you using the UI?

**Comment by smartsammler** [comment]
No, I did not manage to get `wx` working properly on my system (Fedora) when I tested it back then.
And from then on I used a version with the UI parts commented out.
I'll give wx a new try, because with the new version there are even wheels available.

**Comment by zhangt58** [comment]
After PR #13 being merged into master, Python 3.x would be fully supported, including UI (though not quite useful). `pip install beamline --upgrade` will install version `2.0.0`, which requires `wxpython 4.0.1` and `Python 3.x`.

**Comment by zhangt58** [comment]
>No, I did not manage to get wx working properly on my system (Fedora) when I tested it back then.
And from then on I used a version with the UI parts commented out.
I'll give wx a new try, because with the new version there are even wheels available.

@smartsammler I've put a file named `install_deps.sh` in the project directory, which is required before installing `wxpython 4.0.1` via `pip install`, the package names in Fedora might be different.

**Comment by smartsammler** [comment]
Thanks. I'll test it when I'm back on my main system.
